pkg/util/discovery/server_resourcer.go

```diff
@@ -64,6 +65,11 @@ func GetResources(discoveryClient ServerResourcer) ([]*metav1.APIResourceList, s
 			for gv, subErr := range discoErr.Groups {
 				if apierrors.IsNotFound(subErr) {
 					klog.Warningf("Failed to discover APIGroups %s due to not found, moving on: %v", gv, subErr)
+				} else if strings.Contains(subErr.Error(), "received empty response") {
+					// Allow "empty response" as common responses for metrics
+					// Upstream workaround https://github.com/kubernetes/kubernetes/pull/115978
+					// TODO b/349378765 change to type match when emptyResponseError in upstream is made public
+					klog.Warningf("Failed to discover APIGroups %s due to empty response received, moving on: %v", gv, subErr)
```

Comment:
I think you can drop the "due to empty response received" here, since you're including the text of the error. How about `Failed to discover resources for API group %s: assuming zero resources: %v`

Reply:
Done